feat(auth): Creation & Authentication of app

xorima opened the pull request:
# Description

Creation of the action
Setup the authentication for the app

## Issues Resolved

Resolves #5, #7, #8, #9, #10

## Check List

- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

xorima committed:
feat(auth): Creation & Authentication of app

Creation of the action
Setup the authentication for the app
TODO: write tests for this for future
TODO: Setup build & release workflows

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
ci: Cleanup CI files that came with template

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
style(linter): resolve failing linters

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

damacus commented on .github/workflows/linter.yml:
```diff
@@ -42,3 +39,5 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           TYPESCRIPT_DEFAULT_STYLE: prettier
           VALIDATE_JSCPD: false
+          VALIDATE_GITLEAKS: false # false positives atm around cert header
```
Is there any way to configure this warning rather than ignoring Git leaks?

xorima committed:
ci(gitleaks): enable gitleaks to run again

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
feat(tests): Add tests with full coverage

In order to maintain this going forward we need solid tests which
this provides

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
feat(dev): Add Pre-commit hooks & Readme

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
feat(gitleaks): Setup as standalone check

GitLeaks should be a standalone check as the one in the larger linter
does not seem to work correctly with config files

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

xorima committed:
feat(package): Update package name and group

Signed-off-by: Jason Field <jason@avon-lea.co.uk>
